test/collection_test.js

```diff
@@ -0,0 +1,653 @@
+
+      it("should resolve with imported deletions", () => {
+        return articles.pullChanges(result)
+          .then(res => res.deleted)
+          .should.eventually.become([
+            {id: 4}
+          ]);
+      });
+
+      it("should resolve with no conflicts detected", () => {
+        return articles.pullChanges(result)
+          .then(res => res.conflicts)
+          .should.eventually.become([]);
+      });
+
+      it("should actually import changes into the collection", () => {
+        return articles.pullChanges(result)
+          .then(_ => articles.list())
+          .then(res => res.data)
+          .should.eventually.become([
+            {id: 1, title: "art1", _status: "synced"},
+            {id: 2, title: "art2", _status: "synced"},
+            {id: 3, title: "art3", _status: "synced"},
+            {id: 5, title: "art5", _status: "synced"},
+          ]);
+      });
+
+      it("should skip already locally deleted data", () => {
+        return articles.create({title: "foo"})
+          .then(res => articles.delete(res.data.id))
+          .then(res => articles.importChange({id: res.data.id, deleted: true}))
+          .then(res => res.data.title)
+          .should.eventually.become("foo");
+      });
+    });
+
+    describe("When a conflict occured", () => {
+      var createdId;
+
+      beforeEach(() => {
+        return articles.create({title: "art2"})
+          .then(res => {
+            createdId = res.data.id;
+            sandbox.stub(Api.prototype, "fetchChangesSince").returns(
+              Promise.resolve({
+                lastModified: 42,
+                changes: [
+                  {id: createdId, title: "art2mod"}, // will conflict with unsynced local record
+                ]
+              }));
+          });
+      });
+
+      it("should resolve listing conflicting changes", () => {
+        return articles.pullChanges(result)
+          .should.eventually.become({
+            ok: false,
+            lastModified: 42,
+            errors:    [],
+            created:   [],
+            updated:   [],
+            deleted:   [],
+            skipped:   [],
+            published: [],
+            conflicts: [{
+              type: "incoming",
+              local: {
+                _status: "created",
+                id: createdId,
+                title: "art2",
+              },
+              remote: {
+                id: createdId,
+                title: "art2mod",
+              }
+            }]});
+      });
+    });
+
+    describe("When a resolvable conflict occured", () => {
+      var createdId;
+
+      beforeEach(() => {
+        return articles.create({title: "art2"})
+          .then(res => {
+            createdId = res.data.id;
+            sandbox.stub(Api.prototype, "fetchChangesSince").returns(
+              Promise.resolve({
+                lastModified: 42,
+                changes: [
+                  {id: createdId, title: "art2"}, // resolvable conflict
```

Comment:
Automatic reconciliation of perfectly matching records (https://github.com/mozilla-services/cliquetis/blob/first-sync-impl/src/collection.js#L321)